boa/src/object/internal_methods.rs

```diff
@@ -292,34 +455,32 @@ impl GcObject {
     /// [spec]: https://tc39.es/ecma262/#sec-ordinary-object-internal-methods-and-internal-slots-setprototypeof-v
     /// [mdn]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/setPrototypeOf
     #[inline]
-    pub fn set_prototype_of(&mut self, _val: Value) -> bool {
-        // debug_assert!(val.is_object() || val.is_null());
-        // let current = self.prototype.clone();
-        // if same_value(&current, &val) {
-        //     return true;
-        // }
-        // if !self.is_extensible() {
-        //     return false;
-        // }
-        // let mut p = val.clone();
-        // let mut done = false;
-        // while !done {
-        //     if p.is_null() {
-        //         done = true
-        //     } else if same_value(&Value::from(self.clone()), &p) {
-        //         return false;
-        //     } else {
-        //         let prototype = p
-        //             .as_object()
-        //             .expect("prototype should be null or object")
-        //             .prototype
-        //             .clone();
-        //         p = prototype;
-        //     }
-        // }
-        // self.prototype = val;
-        // true
-        todo!("Object.setPropertyOf(obj, prototype)")
+    pub fn set_prototype_of(&mut self, val: Value) -> bool {
+        debug_assert!(val.is_object() || val.is_null());
```

Comment:
Should we assert this also in production builds?

Reply:
https://tc39.es/ecma262/#assert
The spec says:
> Such assertions add no additional semantic requirements and hence need not be checked by an implementation.

So I agree with this approach, I think generally we either don't do it because Rust guarantees the that the assertions are true or we can use `debug_assert!` since that will be found by tests. (Thinking about it, it won't be found by `boa_tester`, right? Since that runs in release mode.)

Reply:
Sure, it can only be detected in unit tests, as running `boa_tester` in debug mod is far too slow.

Reply:
We have some places where the assertions are run in release mode too. I guess it's fine as a debug assertion though.